File: fastapi-brain/app/routes/health.py
from fastapi import APIRouter, Request
from fastapi.responses import JSONResponse

router = APIRouter()


@router.get("/health")
async def health() -> dict:
    return {"status": "ok", "service": "fastapi-brain"}


@router.get("/ready")
async def ready(request: Request) -> JSONResponse:
    try:
        db = request.app.state.db
        await db.query_raw("SELECT 1")
        return JSONResponse({"status": "ok"})
    except Exception:
        return JSONResponse({"status": "unavailable", "reason": "database"}, status_code=503)

File: node-gateway/src/routes/health.ts
import type { FastifyPluginAsync } from 'fastify';
import { prisma } from '../lib/prisma.js';

const healthRoutes: FastifyPluginAsync = async (fastify) => {
  fastify.get('/health', async () => {
    return {
      status: 'ok',
      service: 'node-gateway',
      timestamp: new Date().toISOString(),
    };
  });

  fastify.get('/ready', async (_req, reply) => {
    try {
      await prisma.$queryRaw`SELECT 1`;
      return { status: 'ok' };
    } catch {
      return reply.status(503).send({ status: 'unavailable', reason: 'database' });
    }
  });
};

export default healthRoutes;
